zio-http/shared/src/main/scala/zio/http/endpoint/openapi/JsonSchema.scala

```diff
@@ -552,8 +577,18 @@ object JsonSchema {
       case Schema.Fail(_, _)             => throw new IllegalArgumentException("Fail schema is not supported.")
       case Schema.Tuple2(left, right, _) => AllOfSchema(Chunk(fromZSchema(left, refType), fromZSchema(right, refType)))
       case Schema.Either(left, right, _) => OneOfSchema(Chunk(fromZSchema(left, refType), fromZSchema(right, refType)))
-      case Schema.Lazy(schema0)          => fromZSchema(schema0(), refType)
-      case Schema.Dynamic(_)             => AnyJson
+      case Schema.Fallback(left, right, true, _) =>
+        OneOfSchema(
+          Chunk(
+            AllOfSchema(Chunk(fromZSchema(left, refType), fromZSchema(right, refType))),
+            fromZSchema(left, refType),
+            fromZSchema(right, refType),
+          ),
+        )
+      case Schema.Fallback(left, right, _, _)    =>
+        AnyOfSchema(Chunk(fromZSchema(left, refType), fromZSchema(right, refType)))
```

Comment:
I am not sure this should be any of. Since the flag for full decode is false here, we expect exactly one value.
That would fit more to one of. It is true, that we don't know, if left contains right or the other way around.
But this would also make this API unusable.
Imagine we have a `case class User(name: String)` and `case class Admin(name: String, roles:List[String]`, if you hand over a json with `name` and `roles`, which one will be deserialized (additional fields are allowed by default)? We have here no discriminator. Only structure. If we use one of and one contains the other, at least the error is visible.

Reply:
Ok, I'll change this.

Off the topic, but one thing I'm worried of is the spec doesn't guarantee that the order of elements in `oneOf` or `anyOf` matters, so we can't be 100% sure that it exactly expresses the purpose of `Fallback`.